langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/FunctionCallHelperTest.java

```diff
@@ -0,0 +1,125 @@
+package dev.langchain4j.model.vertexai;
+
+import com.google.cloud.vertexai.api.FunctionCall;
+import com.google.cloud.vertexai.api.FunctionDeclaration;
+import com.google.cloud.vertexai.api.Schema;
+import com.google.cloud.vertexai.api.Tool;
+import com.google.cloud.vertexai.api.Type;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.NullValue;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import dev.langchain4j.agent.tool.JsonSchemaProperty;
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+
+import static dev.langchain4j.model.vertexai.FunctionCallHelper.unwrapProtoValue;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FunctionCallHelperTest {
+    @Test
+    void should_unwrap_proto_values() {
+        // check basic values
+        assertThat(unwrapProtoValue(Value.newBuilder().setStringValue("hello").build())).isEqualTo("hello");
+        assertThat(unwrapProtoValue(Value.newBuilder().setBoolValue(false).build())).isEqualTo(false);
+        assertThat(unwrapProtoValue(Value.newBuilder().setNumberValue(1.23).build())).isEqualTo(1.23);
```

Comment:
Great remark!
I went with only `String`, `Boolean` and `Number` because the underlying gRPC API only supports `Number`s.
There's no distinct type for `int`, `long` and `float`.
A number is just a `double`.
I'm not sure how to proceed there?

Reply:
But aren't `Integer`, `Long`, `Float`, etc. `Number`s?

Would be nice to define a tool with all common types (including float, double, int, long, etc) and see if deserialization of Gemini tool invocation works well and the tool can be called without errors